modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -219,6 +219,9 @@ public void processOpts() {
             importMapping.remove("DateTime");
             instantiationTypes.put("array", "java.util.ArrayList");
             instantiationTypes.put("map", "java.util.HashMap");
+        } else {
+            languageSpecificPrimitives.add("List");
+            languageSpecificPrimitives.add("Map");
```

Comment:
[Without this addition](https://github.com/swagger-api/swagger-codegen/issues/1255#issuecomment-187125334) I got `ListDto<...>` (and I guess also `MapDto<...>`) generated instead of `List<...>` (and `Map<...>`), because toModelName is called from JavaClientCodegen.getSwaggerType, if the type is not a primitive one (or contains a dot).